Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1020,6 +1020,24 @@ extension BrowserViewController: BrowserToolbarDelegate {
         tabManager.selectedTab?.reload()
     }
 
+    func browserToolbarDidLongPressReload(browserToolbar: BrowserToolbarProtocol, button: UIButton) {
+        if #available(iOS 9.0, *) {
+            guard let tab = tabManager.selectedTab, let webView = tab.webView, let currentItem = webView.backForwardList.currentItem else {
+                return
+            }
+
+            let defaults = NSUserDefaults(suiteName: AppInfo.sharedContainerIdentifier())!
+            webView.customUserAgent = UserAgent.desktopUserAgent(defaults)
+
+            if currentItem.URL != currentItem.initialURL {
+                // Reload the initial URL to avoid UA specific redirection
+                tab.loadRequest(NSURLRequest(URL: currentItem.initialURL, cachePolicy: .ReloadIgnoringLocalAndRemoteCacheData, timeoutInterval: 60))
```

Comment:
I chose the standard timeout of `NSURLRequest` which, according to the [docs](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Classes/NSURLRequest_Class/#//apple_ref/occ/clm/NSURLRequest/requestWithURL:), is 60 seconds.